cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -563,15 +563,14 @@ private MetadataEntry convertRow(Row row) {
   public SearchResults search(String namespaceId, String searchQuery, Set<EntityTypeSimpleName> types,
                               SortInfo sortInfo, int offset, int limit, int numCursors,
                               @Nullable String cursor, boolean showHidden) {
-    if (SortInfo.DEFAULT.equals(sortInfo)) {
-      return searchByDefaultIndex(namespaceId, searchQuery, types, showHidden);
-    }
-    return searchByCustomIndex(namespaceId, types, sortInfo, offset, limit, numCursors, cursor, showHidden);
+    return SortInfo.DEFAULT.equals(sortInfo) ?
+      searchByDefaultIndex(namespaceId, searchQuery, types, showHidden) :
+      searchByCustomIndex(namespaceId, types, sortInfo, offset, limit, numCursors, cursor, showHidden);
   }
 
   private SearchResults searchByDefaultIndex(String namespaceId, String searchQuery,
                                              Set<EntityTypeSimpleName> types, boolean showHidden) {
-    List<MetadataEntry> results = new ArrayList<>();
+    List<MetadataEntry> results = new LinkedList<>();
```

Comment:
Since we're not doing any random-access and we don't know the size of the results prior to creation, I feel like it would be more efficient to use a `LinkedList` over an `ArrayList`. Please let me know if I'm wrong about this.